consoles/sshVirtsh.pm

```diff
@@ -325,111 +326,126 @@ sub add_interface {
+        if ($stderr) {
+            bmwqemu::diag("gen_kvm_img failed: $stderr, try $_ out of $tries");
+            sleep 5;
         }
+        last unless $ret;
     }
-    else {    # Copy image to VM host
-        die 'No file given' unless $args->{file};
-        my $file_basename             = basename($args->{file});
-        my $vmware_disk_path          = $vmware_openqa_datastore . $file_basename;
-        my $vmware_disk_path_thinfile = $vmware_disk_path =~ s/\.vmdk/_${name}_thinfile\.vmdk/r;
-        if ($cdrom || $backingfile) {
-            if ($self->vmm_family eq 'vmware') {
-                # If the file exists, make sure someone else is not copying it there right now,
-                # otherwise copy image from NFS datastore.
-                my $nfs_dir              = $backingfile ? 'hdd' : 'iso';
-                my $vmware_nfs_datastore = get_required_var('VMWARE_NFS_DATASTORE');
-                my $cmd =
-                  "if test -e $vmware_openqa_datastore$file_basename; then " .
-                  "while lsof | grep 'cp.*$file_basename'; do " .
-                  "echo File $file_basename is being copied by other process, sleeping for 60 seconds; sleep 60;" .
-                  'done;' .
-                  'else ' .
-                  "cp /vmfs/volumes/$vmware_nfs_datastore/$nfs_dir/$file_basename $vmware_openqa_datastore;" .
-                  'fi;';
-                my $retval = $self->run_cmd($cmd, domain => 'sshVMwareServer');
-                die "Can't copy VMware image $file_basename" if $retval;
-                if ($backingfile) {
-                    # Power VM off, delete it's disk image, and create it again.
-                    # Than wait for some time for the VM to *really* turn off.
-                    my $cmd =
-                      "( set -x; vmid=\$(vim-cmd vmsvc/getallvms | awk \'/$name/ { print \$1 }\');" .
-                      'if [ $vmid ]; then ' .
-                      'vim-cmd vmsvc/power.off $vmid;' .
-                      'fi;' .
-                      "vmkfstools -v1 -U $vmware_disk_path_thinfile;" .
-                      "vmkfstools -v1 -i $vmware_disk_path --diskformat thin $vmware_disk_path_thinfile; sleep 10 ) 2>&1";
-                    my $retval = $self->run_cmd($cmd, domain => 'sshVMwareServer');
-                    die "Can't create thin VMware image" if $retval;
-                }
-            }
-            else {
-                $self->run_cmd(sprintf("rsync -av '$args->{file}' '$basedir/%s'", $file_basename)) && die 'rsync failed';
-                if ($file_basename =~ /(.*)\.xz$/) {
-                    $self->run_cmd(sprintf("nice ionice unxz -f -k '$basedir/%s'", $file_basename)) unless -e "$basedir$1";
-                    $file_basename = $1;
-                }
-            }
+    die "Too many attempts to format HDD" if $stderr;
+    bmwqemu::diag("gen_kvm_img: Finished creating image file...");
+}
+
+sub download_image_file {
+    my ($self, $src, $dest) = @_;
```

Comment:
I would add here some description... That the destination must be given as path or so...